### PR TITLE
Add linting

### PR DESCRIPTION
**Pull request by martinheidegger**
Trying to submit other Pull Requests I tried finding out what sort of code style you prefer. I didn't quite figure it out but the closest I could come was `semistandard` with [a few changes](https://github.com/song940/node-dns/compare/master...martinheidegger:lint?expand=1#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822R4-R14).

The linter also found a few problem that should be fixed with this PR:

- `query` is used before assignment [link](https://github.com/song940/node-dns/compare/master...martinheidegger:lint?expand=1#diff-64f6ddc01f0d332f6f205fdc4464f4dafb9da32b7737650088a55c90a9bfa344L32-L36)
- `query` is used but not defined [link#1](https://github.com/song940/node-dns/compare/master...martinheidegger:lint?expand=1#diff-547ee4ace210b75f8f4c0ae1038db39bad8b6a68b2e387273f588e0b91dd58c0L10) [link#2](https://github.com/song940/node-dns/compare/master...martinheidegger:lint?expand=1#diff-547ee4ace210b75f8f4c0ae1038db39bad8b6a68b2e387273f588e0b91dd58c0L13)
- `url.parse` is deprecated [link](https://github.com/song940/node-dns/compare/master...martinheidegger:lint?expand=1#diff-f823a629b715f7a0218b7c1441a489e6700aa203d7aad277385b164afa0d3c6eL3)

### CODE REVIEW

File: client/doh.js [hunk 1]
```diff
@@ -29,23 +29,23 @@ const DOHClient = ({ dns } = {}) => {
     if (recursive) {
       packet.header.rd = 1;
     }
-    if (clientIp) {
-      query.additionals.push(Packet.Resource.EDNS([
```
Comment (song940) [inline client/doh.js:33]: Rename “query” to “packet”

File: client/doh.js [hunk 2]
```diff
@@ -29,23 +29,23 @@ const DOHClient = ({ dns } = {}) => {
...
-      type: Packet.TYPE[type],
+      class : cls,
+      type  : Packet.TYPE[type],
     });
     const query = packet.toBase64URL();
+    if (clientIp) {
+      query.additionals.push(Packet.Resource.EDNS([
+        Packet.Resource.EDNS.ECS(clientIp),
+      ]));
+    }
```
Comment (song940) [inline client/doh.js:42]: Remove this block, because “query” is String

**Review by song940 — COMMENTED** [review]
Thank for your good work, I will checking it as soon as possible.

**Comment by song940** [comment]
Missing “DevDependencies“ in package.json .

File: client/doh.js (hunk 2 above)
Comment (martinheidegger) [reply client/doh.js:42]: Updated both mentions, can you check the block?

**Comment by song940** [comment]
Looks good to me.

**Pull request merged**